src/content/en/2021/cdn.md

```diff
@@ -17,4 +17,351 @@ featured_stat_3: TODO
+
+These are the limits to our testing methodology:
+
+**Simulated network latency:** We use a dedicated network connection that synthetically shapes traffic.
+
+**Single geographic location:** Tests are run from a single datacenter and cannot test the geographic distribution of many CDN vendors.
+
+**Cache effectiveness:** Each CDN uses proprietary technology and many, for security reasons, do not expose cache performance.
+
+**Localization and internationalization: **Just like geographic distribution, the effects of language and geo-specific domains are also opaque to these tests.
+
+**CDN detection:** This is primarily done through DNS resolution and HTTP headers. Most CDNs use a DNS CNAME to map a user to an optimal datacenter. However, some CDNs use AnyCast IPs or direct A+AAAA responses from a delegated domain which hide the DNS chain. In other cases, websites use multiple CDNs to balance between vendors, which is hidden from the single-request pass of WebPageTest. 
+
+All of this limits the effectiveness in the measurements.
+
+Most importantly, these results reflect the utilization of specific features (Example: TLS, HTTP/2 etc.,) but do not reflect actual impact. YouTube is more popular than “www.example.com” yet both will appear as equal value when comparing utilization.
+
+With this in mind, here are a few statistics that were intentionally not measured in the context of a CDN:
+
+
+
+1. Time To First Byte(TTFB)
+2. CDN Round Trip Time
+3. Core Web Vitals
+4. Cache Hit v/s Cache Miss performance etc.,
+
+**CDN Adoption**
+
+The contents in a web page can be divided into 3 parts, namely:
+
+
+
+1. Base HTML Page (Ex: [www.example.com](www.example.com))
+2. Embedded First Party Content (Ex: images.example.com, css.example.com etc.,)
+3. 3rd Party Content (Ex: Google Analytics, Advertisements etc.,)
+
+From their inception, CDNs have been the go-to solution for delivering embedded content like: images, stylesheets, javascripts, fonts, etc.  This kind of  content doesn't change frequently, making it a good candidate for caching on a CDN’s proxy servers.
+
+With the evolution of CDN technology an expressway was set up on the internet for non-cacheable assets.  This means HTMLs and APIs can now be delivered reliably and faster, compared to the standard TCP connection.
+
+
+
+<p id="gdcalert1" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image1.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert2">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>
+
+
+![alt_text](images/image1.png "image_tooltip")
```

Comment:
Follow the steps in the [Figures Guide](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) to get the charts embedded in the markdown